workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -90,6 +90,26 @@ private static void LaunchAndroid(bool shouldConnectLocally)
                 else
                 {
                     arguments.Append($"+{RuntimeConfigNames.Environment} {RuntimeConfigDefaults.CloudEnvironment} ");
+
+                    var gdkToolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
+
+                    // Return error if no DevAuthToken is set AND fails to generate new DevAuthToken.
+                    if (!EditorPrefs.HasKey(RuntimeConfigNames.DevAuthTokenKey) && !DevAuthTokenUtils.Generate())
+                    {
+                        Debug.LogError("Failed to generate a Dev Auth Token to launch mobile client.");
+                        return;
+                    }
+
+                    // Returns error if DevAuthToken.txt does not exists AND fails to save new DevAuthToken.txt.
+                    // Ignores if GdkToolsConfiguration does not require DevAuthToken to be saved as text file
+                    if (gdkToolsConfig.SaveDevAuthTokenToFile &&
+                        (!File.Exists(gdkToolsConfig.DevAuthTokenFilepath) || !DevAuthTokenUtils.SaveTokenToFile()))
```

Comment:
by this point it's too late to save the file

remove the .txt creation but include the file check + print a warning